src/interfaces/records-read.ts

```diff
@@ -57,6 +75,28 @@ export class RecordsRead extends Message<RecordsReadMessage> {
     return new RecordsRead(message);
   }
 
+  /**
+   * Creates a Records interface Filter using either the `recordId` or `protocol` & `protocolPath` of the incoming Message.
+   * @param descriptor message descriptor with optional properties `recordId`, `protocol` and `protocolPath`
+   *
+   * @returns {Filter} with a Records interface as well as the appropriate filter params
+   * @throws {DwnError} when either `recordId` is missing, or both `protocol` and `protocolPath` are missing.
+   */
+  public static createFilter(descriptor: RecordsReadDescriptor): Filter {
+    const commonFilter: Filter = { interface: DwnInterfaceName.Records, isLatestBaseState: true };
+    const { recordId, protocol, protocolPath } = descriptor;
+    if (recordId !== undefined) {
+      return { ...commonFilter, recordId };
+    } else if (protocol !== undefined && protocolPath !== undefined) {
+      return { ...commonFilter, protocol, protocolPath };
+    } else {
+      throw new DwnError(
+        DwnErrorCode.RecordsReadMissingDescriptorProperties,
+        'missing required properties from RecordsRead descriptor, expected either `recordId`, or both `protocol` and `protocolPath` to be present'
+      );
+    }
```

Comment:
Sanity: don't think we need this check, as JSON schema check would have caught this condition prior to this function call, so this is a condition only triggered by test.

Consider removing this condition, or if you prefer this for defensive programming, rename the error to something like `DwnErrorCode.RecordsReadCreateFilterUnreachableCode`

Reply:
Hey @LiranCohen, yeah I understand the intent, but as it was, it was just for tests to hit the code, the actual running of DWN would never hit it thus I removed it.

As mentioned above, if you want to have the check in the same place, better to rename the error. If you want to move it so that it is actually called at runtime, then totally open to hear your proposal, but I suspect it will be a much more involved discussion.